app/Services/EntityService.php

```diff
@@ -657,13 +656,15 @@ public function makeNewMentionEntity(MiscModel $model, string $name)
         $model->forceSavedObserver = true;
         $model->is_private = $defaultPrivate;
         $model->save();
-        if ($model->entity->type() != 'tag') {
-            $tags = \App\Models\Tag::autoApplied()->get();
+        if ($model->entity->type_id !== config('entities.ids.tag')) {
+            $allTags = [];
+            $tags = \App\Models\Tag::autoApplied()->with('entity')->get();
```

Comment:
Nice improvement, but this loop is still executed for every new mention (fetching the tags). So if I have 100 `[new:character:ABC]` in the text, it will execute 100 times `App\Models\Tags::get()`.

Can you think of a way to improve this by doing the request only once? Maybe saving the values in a protected variable on `EntityService`? If you need inspiration, check out `EntityService::newEntityTypes()`